### DISCUSSION TITLE
How can I enable returning(*) for Postgres?

### DISCUSSION THREAD

**Discussion by awanuch** [feathersjs-ecosystem/feathers-objection#167] [Q&A] [answered]
Hi,

I'm trying to do inserts and updates that return the full record, not just the fields that are passed in the request.
Objection docs say to run either a InsertAndFetch, or to add returning('*') to the query. Is this supported by the Feathers Objection adaptor? Or is there a better pattern to post/patch the record and return the entire resulting record?

Thank you in advance!

**Comment by dekelev** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/4444781/150629800-ea9dc47a-c9d1-4f4a-8875-07f5e2cac749.png)

By default create fetches the inserted record/s from DB. You can use `$noSelect` for increased performance when needed.

**Reply by awanuch** [reply]
You're correct, thanks for pointing that out. The problem was that I was using a composite ID along with knexSnakeCaseMappers, however in this particular model I had defined the idColumn in snake case instead of camel case. When I ran a create, the only result I was getting back was the composite ID columns instead of the whole record. Changing the idColumns to the correct casing fixed the issue.

My mistake!
